fix compile with Qt6 +tf2 headers deprecation

mosfet80 opened the pull request:
fix compile with Qt6
Tf2 headers  deprecation

mosfet80 committed:
Qt6 fix

Update CMakeLists.txt

Replace ament_target_dependencies with target_link_libraries

Signed-off-by: Alejandro Hernandez Cordero <ahcorde@gmail.com>
Update CMakeLists.txt
Remove Qt5::Widgets from target link libraries
Fix indentation in CMakeLists.txt for Qt version check
update package.zml

Signed-off-by: Alejandro Hernandez Cordero <ahcorde@gmail.com>

Update CMakeLists.txt

Co-authored-by: Nathan Brooks <nbbrooks@gmail.com>
Fix conditional check for Qt version

mosfet80 commented:
Close #288

mosfet80 commented:
@tylerjw ping

mosfet80 commented:
@nbbrooks ping

nbbrooks commented:
While resolving a conflict I've added a link libraries list pattern to reduce the redundant listing.

nbbrooks commented on CMakeLists.txt:
```diff
@@ -40,7 +40,21 @@ find_package(rviz_common REQUIRED)
+	endif()
+else()#Qt5
+	find_package(Qt${QT_VERSION_MAJOR} REQUIRED COMPONENTS Test Widgets)
+endif()
```
```suggestion
find_package(QT NAMES Qt6 Qt5 REQUIRED COMPONENTS Test  Widgets)
find_package(Qt${QT_VERSION_MAJOR} REQUIRED COMPONENTS Widgets)
if(${QT_VERSION} VERSION_LESS 5.15.0)
  function(qt_wrap_cpp out)
    qt5_wrap_cpp(_sources ${ARGN})
    set("${out}" ${_sources} PARENT_SCOPE)
  endfunction()
endif()
if(Qt${QT_VERSION_MAJOR} STREQUAL "6")
  find_package(Qt${QT_VERSION_MAJOR} REQUIRED COMPONENTS Core5Compat)
endif()
```

CI problem might be here. `if(${QT_VERSION} VERSION_LESS 5.15.0)` is inside the `if(Qt${QT_VERSION_MAJOR} STREQUAL "6")` block.

nbbrooks replied:
Adding the explicit check might just move the failure to find_package time, as we have nothing in package.xml to selectively install stuff like `libqt6-core` and `libqt6core5compat6-dev`

I don't know enough about qt5 and qt6 support to know what should get changed there though.

nbbrooks replied:
Nevermind about the explciit check, I do see it is in `find_package(Qt${QT_VERSION_MAJOR} REQUIRED COMPONENTS Test Widgets Core5Compat)` already

nbbrooks replied:
But it is inside the incorrect `if(Qt${QT_VERSION_MAJOR} STREQUAL "6")` syntax. OK let's see what my suggested change does. Prob move the error to find_package time. And then we may need to figure out how to support selective package.xml dependencies based on ROS version (a topic that came up in the [December community meeting](https://discourse.openrobotics.org/t/moveit-community-meeting-thursday-dec-18/51312/3))

simonschmeisser commented:
In ROS 1 branch this followed the Qt version exported by RViz, is this no longer available here?

mosfet80 commented:
@nbbrooks  fixed!

mosfet80 committed:
fix compile with qt6

mosfet80 committed:
fix compile with qt6

Clean up key_tool.hpp by removing unused code

Removed commented-out preprocessor directives and includes.
Clean up preprocessor directives in rviz_visual_tools_gui.hpp

Removed unnecessary preprocessor directives for rclcpp.
Update checkout action version from v4 to v6
Update industrial_ci_action.yaml

mosfet80 commented:
@nbbrooks It works! tested on opensuse 16

mosfet80 committed:
Update .pre-commit-config.yaml

mosfet80 committed:
Update checkout action version in format workflow

mosfet80 committed:
Upgrade GitHub Actions to latest versions

mosfet80 committed:
Downgrade actions/cache from v7 to v6
